config/common.go

```diff
@@ -140,3 +156,75 @@ func (cs *commonStore) validate(cfg *model.Config) error {
 
 	return nil
 }
+
+// mergeConfig merges two configs together. The receiver's values are overwritten with the patch's
+// values except when the patch's values are nil.
+func (cs *commonStore) mergeConfig(patch *model.Config) (*model.Config, error) {
+	ret, err := utils.Merge(cs.config, patch)
+	if err != nil {
+		return nil, err
+	}
+	retC := ret.(model.Config)
+	return &retC, nil
+}
+
+// removeEnvOverrides takes the newCfg provided and adds information stored in the commonStore,
+// then delegates the task to the removeEnvOverrides function
+func (cs *commonStore) removeEnvOverrides(newCfg *model.Config) *model.Config {
+	return removeEnvOverrides(newCfg, cs.configWithoutOverrides, cs.environmentOverrides)
+}
+
+// removeEnvOverrides returns a new config without the current environment overrides.
```

Comment:
`s/current/given`

Reply:
Also, propose moving this and below into another file, perhaps `environment.go`?

Reply:
Good idea. Is there a MM style guide to help with deciding when to split off functions, and how (eg, preferred length of files)? 

For this, I put all the functions in `environment.go` but kept the commonStore method in `common.go`.

Reply:
There probably /should/ be a style guide for this, but I'm honestly just using the force here. It feels like the environment stuff forms a cluster of helper methods that are relatively independent from the common store, and so it might be easier to browse through them in a separate file. It's an incredibly subjective opinion, and I'm 0/5 on it :p